R/quanteda_options.R

```diff
@@ -155,6 +155,6 @@ get_options_default <- function(){
                  base_compname = "comp",
                  language_stemmer = "english",
                  pattern_hashtag = "#\\w+#?",
-                 pattern_username = "@")
+                 pattern_username = "@[a-zA-Z0-9]+")
```

Comment:
That's not quite precise enough. 
``` r
patt1957 <- "@[a-zA-Z0-9]+"
pattbetter <- "^@[a-zA-Z0-9_]+$"

txt <- c("kbenoit@lse.ac.uk", "@@invalid", "@kenbenoit", "@4ever", "@ken_ben")

stringi::stri_detect_regex(txt, patt1957)
## [1] TRUE TRUE TRUE TRUE TRUE
stringi::stri_detect_regex(txt, pattbetter)
## [1] FALSE FALSE  TRUE  TRUE  TRUE
```
Twitter rules are here https://help.twitter.com/en/managing-your-account/twitter-username-rules and allow just 15 characters, so we could even say:
```r
"^@[a-zA-Z0-9_]{1,15}$"
```
if we wanted to stick to Twitter.

There is an accepted answer [on SO](https://stackoverflow.com/questions/2304632/regex-for-twitter-username/6351873) that is more complicated, but I think it's based on when Twitter did not permit the first character of the username to be a number. (That's permitted now.)

Reply:
This is not for tokens, not need of `^` and `$` but added `_`.

Reply:
Internet address will be captured by `address <- "(https?:\\/\\/(www\\.)?|@)[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-z]{2,4}\\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)"` before the username matching in `preserve_special()`.

Reply:
OK, as long as it works (wherever it's used) correctly for the test cases above.